scikeras/wrappers.py

```diff
@@ -633,28 +707,49 @@ def score(self, X, y, sample_weight=None, **kwargs):
         y = check_array(y, ensure_2d=False)
 
         # compute Keras model score
-        y_pred = self.predict(X, **kwargs)
+        y_pred = self.predict(X)
+
+        # filter kwargs and get attributes for score
+        params = self.get_params()
+        pred_args = route_params(
+            params, destination="score", pass_filter=set()
+        )
 
-        return self._scorer(y, y_pred, sample_weight=sample_weight)
+        return self.scorer(y, y_pred, sample_weight=sample_weight)
 
-    def _filter_params(self, fn, params_to_check=None):
-        """Filters all instance attributes (parameters) and
-             returns those in `fn`'s arguments.
+    def get_meta_params(self) -> Dict[str, Any]:
+        """Get meta parameters (parameters created by fit, like
+        n_features_in_ or target_type_).
 
-        Arguments:
-            fn : arbitrary function
-            params_to_check : dictionary, parameters to check.
-                Defaults to checking all attributes of this estimator.
+        Returns
+        -------
+        Dict[str, Any]
+            Dictionary of meta parameters
+        """
+        return {
+            k: v
+            for k, v in self.__dict__.items()
+            if (
+                k not in type(self)().__dict__
+                and k not in self.get_params()
+                and (k.startswith("_") or k.endswith("_"))
```

Comment:
Suggested change:

``` diff
- and (k.startswith("_") or k.endswith("_"))
+ and (k.endswith("_") and not k.startswith("_"))
```

Why are we showing private attributes to the user?

Reply:
Yeah I think that is better. I don't see why the user would need to access `_random_state` and other params like that from within `model_build_fn`